lib/Rex/Test/Base/has_checksum.pm

```diff
@@ -0,0 +1,54 @@
+#
+# (c) Jan Gehring <jan.gehring@gmail.com>
+#
+# vim: set ts=2 sw=2 tw=0:
+# vim: set expandtab:
+
+package Rex::Test::Base::has_checksum;
+
+use strict;
+use warnings;
+
+# VERSION
+
+use Rex -base;
+use Digest::SHA;
+use Rex::Commands::MD5;
+use base qw(Rex::Test::Base);
+
+sub new {
+  my $that  = shift;
+  my $proto = ref($that) || $that;
+  my $self  = {@_};
+
+  bless( $self, $proto );
+
+  my ( $pkg, $file ) = caller(0);
+
+  return $self;
+}
+
+sub run_test {
+  my ( $self, $file, $checksum, $algo, $computed ) = (shift, shift, shift, shift);
```

Comment:
I'm not entirely sure about using positional arguments here. Perhaps it might be better with a hash?

Also, there are 5 variables on the left hand side, but only 4 `shift`s on the right hand side. Perhaps `$computed` can be dropped since doesn't seem like used as an input argument?